client/my-sites/site-settings/settings-newsletter/main.tsx

```diff
@@ -176,16 +174,11 @@ const NewsletterSettingsForm = wrapSettingsForm( getFormSettings )( ( {
 	} );
 
 	const disabled = isSubscriptionModuleInactive || isRequestingSettings || isSavingSettings;
-	const savedSubscriptionOptions = settings?.subscription_options;
```

Comment:
I'm not sure why this was needed, since the form and wrapper handle updating the settings.